builtin/credential/aws/path_role.go

```diff
@@ -41,14 +54,33 @@ that it must match the IAM role ARN specified by this parameter.
 The value is prefix-matched (as though it were a glob ending in
 '*').  The configured IAM user or EC2 instance role must be allowed
 to execute the 'iam:GetInstanceProfile' action if this is
-specified.`,
+specified. This is only checked when auth_type is
+ec2.`,
 			},
 			"bound_iam_instance_profile_arn": {
 				Type: framework.TypeString,
 				Description: `If set, defines a constraint on the EC2 instances to be associated
 with an IAM instance profile ARN which has a prefix that matches
 the value specified by this parameter. The value is prefix-matched
-(as though it were a glob ending in '*').`,
+(as though it were a glob ending in '*'). This is only checked when
+auth_type is ec2.`,
+			},
+			"role_inferred_type": {
+				Type:    framework.TypeString,
+				Default: false,
+				Description: `When auth_type is iam, the
```

Comment:
I get what you're saying; I wasn't really aware that there might be ambiguities with the instance ID. That's fine, let's leave as-is.